### PR TITLE
Query helper

### PR DESCRIPTION
**Pull request by BearLemma**
First little simplification and quality life improvement for our endpoints.
Introduces Query structure to simplify parsing and working with query string.

### CODE REVIEW

**Review by MarinPostma — COMMENTED** [review]
I there a way to make `get` a generic method?

File: api/src/chisel.ts [hunk 1]
```diff
@@ -678,6 +682,57 @@ export class ChiselRequest extends Request {
     }
 }
 
+/** Query is a helper class used to access query parameters parsed from the URL.
+ */
+export class Query {
```
Comment (honzasp) [inline api/src/chisel.ts:687]: Maybe we should also implement other methods from `URLSearchParams`:

- `entries()`, `keys()`, `values()` and `forEach()` for iteration
- `has()` for existence checking
- `getAll()` to get all occurences of one key (`get()` returns only the first value)

All these methods should directly call the corresponding methods of `params`.

Comment (honzasp) [reply api/src/chisel.ts:687]: Or perhaps we can directly inherit from `URLSearchParams` and get all these methods for free? :thinking:

**Review by honzasp — COMMENTED** [review]
I think this is great, but the API might be improved:
- Add more methods from `URLSearchParams`, such as `getAll()` or `entries()`
- Don't use the explicit default values in the getters, let the caller use `?? default` if they want to (this also makes `get()` compatible with `URLSearchParams`)
- Change the semantics of boolean parsing; at the very least, `"0"` should be treated as false.
(More details in comments)

File: api/src/chisel.ts [hunk 2]
```diff
@@ -678,6 +682,57 @@ export class ChiselRequest extends Request {
...
+     * parameter exists, returns `undefined` or `defaultValue` if provided.
+     * If `paramName` key exists, the value is converted to lowercase and then compared to `'true'`.
+     * This means that the function returns false for anything other than variations of `true` string.
+     * @param paramName query parameter to be retrieved from the URL's query string.
+     * @param defaultValue default value to be returned when `paramKey` is not present.
+     */
+    getBool(paramName: string, defaultValue?: boolean): boolean | undefined {
+        const v = this.get(paramName);
+        if (v !== undefined) {
+            return v.toLowerCase() === "true";
```
[the remark this answers is not in the document]
Comment (honzasp) [reply api/src/chisel.ts:730]: At the very least, we must accept `"1"` as a true value.

File: api/src/chisel.ts (hunk 1 above)
Comment (honzasp) [reply api/src/chisel.ts:687]: You are right, let's just implement these methods without inheriting from `URLSearchParams`.

**Review by honzasp — DISMISSED** [review]

**Review by honzasp — APPROVED** [review]

**Pull request merged**